enterprise-search/utils/retriever.py

```diff
@@ -0,0 +1,107 @@
+"""Retriever wrapper for Google Cloud Enterprise Search."""
+from __future__ import annotations
+
+import json
+from typing import Any, Dict, List, Optional
+
+from google.protobuf.json_format import MessageToDict
+from google.cloud.discoveryengine_v1beta  import SearchServiceClient, SearchRequest
```

Comment:
For readability, it's recommended to import the full cloud library instead of individual elements.